packages/iov-lisk/src/liskclient.ts

```diff
@@ -0,0 +1,200 @@
+import axios from "axios";
+import { ReadonlyDate } from "readonly-date";
+import { Stream } from "xstream";
+
+import {
+  Address,
+  BcpAccount,
+  BcpAccountQuery,
+  BcpAddressQuery,
+  BcpNonce,
+  BcpQueryEnvelope,
+  BcpTicker,
+  BcpTransactionResponse,
+  ConfirmedTransaction,
+  IovReader,
+  Nonce,
+  TokenTicker,
+} from "@iov/bcp-types";
+import { Encoding } from "@iov/encoding";
+import { Algorithm, ChainId, PostableBytes, PublicKeyBytes, Tag, TxId, TxQuery } from "@iov/tendermint-types";
+
+import { Parse } from "./parse";
+
+function isAddressQuery(query: BcpAccountQuery): query is BcpAddressQuery {
+  return (query as BcpAddressQuery).address !== undefined;
+}
+
+/**
+ * Encodes the current date and time as a nonce
+ */
+export function generateNonce(): Nonce {
+  const now = new ReadonlyDate(ReadonlyDate.now());
+  return Parse.timeToNonce(now);
+}
+
+export class LiskClient implements IovReader {
+  private readonly baseUrl: string;
+
+  constructor(baseUrl: string) {
+    this.baseUrl = baseUrl;
+  }
+
+  public disconnect(): void {
+    // no-op
+  }
+
+  public async chainId(): Promise<ChainId> {
+    const url = this.baseUrl + "/api/node/constants";
```

Comment:
Just a note, I had some issues with tendermint rpc, when accidentally entering a baseurl that ended with `/`, so the combination looked like `http://host//path`.

Maybe handle that here as well, eg. trimming trailing '/' in the constructor. Or something safer

Reply:
`http://host//path` is not an issue and should work exactly like `http://host/path`. But we should validate that it is actually a baseUrl including the scheme and not a host like `testnet.list.io` or `testnet.list.io:8000`

Reply:
Oh I was too quick here. `http://host//path` is indeed 404 in the Lisk API. I'l normalize that in addition to the format check